### PR TITLE
[hardreset] script to automate the hardreset

### PR DESCRIPTION
**Pull request by LeoHChen**
Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

File: pipeline/hard-reset.sh [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+function restart_explorer
+{
+   msg "restart explorer backend"
+   explorerip=$(host ${configs[explorer.name]} | awk ' { print $NF } ')
+# TODO: testing
+   echo ./node_ssh.sh ${explorerip} <<EOC
+      tmux new -s explorerbe -d || { tmux kill-session -t explorerbe && tmux new -s explorerbe -d; } ;
+      sleep 1 ;
+      tmux send-keys -t explorerbe "npm run db-reset; sleep 3; npm run start" Enter;
```
Comment (denniswon) [inline pipeline/hard-reset.sh:165]: npm run db-reset will ask for a confirmation and needs user input 'y'.

I will udpate block explorer repo npm command to add `npm run db-reset-confirm` which will do db-rest with auto-confirmation. We can use that for this script instead.
@LeoHChen 
Also, don't need to wait 3 seconds. Please change this to:
`npm run db-reset-confirm; npm run rebuild; npm run start`

File: pipeline/hard-reset.sh [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+function restart_dashboard
+{
+   msg "clear dashboard db"
+   dashboardip=$(host ${configs[dashboard.name]} | awk ' { print $NF } ')
+# TODO: testing
+   echo ./node_ssh.sh ${dashboardip} <<EOC
+      tmux new -s dashboardbe -d || { tmux kill-session -t dashboardbe && tmux new -s dashboardbe -d; } ;
+      sleep 1 ;
+      tmux send-keys -t dashboardbe "cd appengine && npm install" Enter;
```
Comment (denniswon) [inline pipeline/hard-reset.sh:178]: this is incorrect.

Similar to my above comment on block explorer command, let me add npm commands we can use in this script for auto-recovery.
Please update here as:
`cd frontend; npm run db-reset-confirm; cd ../appengine; ...` (... will be cmds for deploying appengine. @john-harmony can fill this in)

**Comment by denniswon** [comment]
https://github.com/harmony-one/staking-dashboard/pull/336
Should update per the scripts added here for staking explorer.

File: pipeline/hard-reset.sh (hunk 2 above)
Comment (denniswon) [reply pipeline/hard-reset.sh:178]: command: please update this to:
```cd frontend; npm run db-reset -- -n [ostn|pstn|stn] (-y); cd ..```

File: pipeline/hard-reset.sh [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+check_version
+{
+   shard=$1
+   msg "./run_on_shard.sh -p ${net_profile} -T $shard './harmony -version'"
+}
+
+restart_watchdog
+{
+   msg "./restart_watchdog.sh -a restart -s $network"
+   ./restart_watchdog.sh -a restart -s "$network"
```
Comment (janet-harmony) [inline pipeline/hard-reset.sh:115]: Are the IP lists assumed to stay the same?
If not, use `-u` to pull latest nodedb before restart.

Comment (LeoHChen) [reply pipeline/hard-reset.sh:115]: for hard-reset, we assume the same ip, not a fresh new network.

**Pull request merged**